lib/connectors/sentinel_connector.js

```diff
@@ -48,21 +48,26 @@ SentinelConnector.prototype.connect = function (callback) {
     if (_this.currentPoint === _this.sentinels.length) {
       _this.currentPoint = -1;
 
+      var error;
       var retryDelay;
       if (typeof _this.options.sentinelRetryStrategy === 'function') {
         retryDelay = _this.options.sentinelRetryStrategy(++_this.retryAttempts);
       }
+
       if (typeof retryDelay !== 'number') {
         debug('All sentinels are unreachable and retry is disabled, emitting error...');
-        var error = 'All sentinels are unreachable.';
-        if (lastError) {
-          error += ' Last error: ' + lastError.message;
-        }
-        return callback(new Error(error));
+        error = 'All sentinels are unreachable.';
+      } else {
+        debug('All sentinels are unreachable. Retrying from scratch after %d', retryDelay);
+        error = 'All sentinels are unreachable, retrying...';
+        setTimeout(connectToNext, retryDelay);
       }
-      debug('All sentinels are unreachable. Retrying from scratch after %d', retryDelay);
-      setTimeout(connectToNext, retryDelay);
-      return;
+
+      if (lastError) {
+        error += ' Last error: ' + lastError.message;
+      }
+      return callback(new Error(error));
```

Comment:
We shouldn't `callback()` with an error here when we still keep retrying since a callback must not be called more than once.

Instead, what about just emit an event (like "sentinelError") when all sentinels are unreachable just like when a connection to a standalone Redis server is down.